### PR TITLE
Clean up plugin

### PR DESCRIPTION
**Pull request by basil**
Removes the following unneeded JARs from the JPI:

commons-io-2.11.0.jar
commons-lang3-3.9.jar
jsr305-3.0.1.jar
slf4j-api-1.7.36.jar

JSR305 is deprecated and the other three are already bundled in Jenkins core, superseding the version bundled in any plugin JPIs.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -30,8 +30,8 @@
         <jenkins.version>2.361.4</jenkins.version>
         <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
         <no-test-jar>false</no-test-jar>
-        <!-- Use the same configuration than before. -->
-        <spotbugs.failOnError>false</spotbugs.failOnError>
+        <!-- TODO fix existing violations -->
+        <spotbugs.threshold>High</spotbugs.threshold>
```
Comment (basil) [inline pom.xml:34]: Make SpotBugs error behavior the same locally as in CI, focusing on high-priority violations.

File: pom.xml [hunk 2]
```diff
@@ -100,29 +100,47 @@
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-cps</artifactId>
         </dependency>
-        <dependency>
-            <groupId>org.jenkins-ci.plugins</groupId>
-            <artifactId>structs</artifactId>
-        </dependency>
```
Comment (basil) [inline pom.xml:106]: Not needed as of recent cores.

File: pom.xml [hunk 3]
```diff
@@ -135,11 +153,6 @@
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>plain-credentials</artifactId>
         </dependency>
-        <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <scope>test</scope>
-        </dependency>
```
Comment (basil) [inline pom.xml:142]: Not needed as of recent plugin parent POMs.

File: src/main/java/com/myyearbook/hudson/plugins/confluence/ConfluencePublisher.java [hunk 4]
```diff
@@ -276,8 +276,9 @@ protected List<Content> performAttachments(Run build, FilePath ws,
                         + " workspace artifact(s) to upload to Confluence...");
 
                 for (FilePath file : workspaceFiles) {
-                    if (!files.contains(file)) {
-                        files.add(file.toVirtualFile());
```
Comment (basil) [inline src/main/java/com/myyearbook/hudson/plugins/confluence/ConfluencePublisher.java:280]: The only high-severity SpotBugs warning identified.

File: src/main/java/com/myyearbook/hudson/plugins/confluence/ConfluenceSite.java [hunk 5]
```diff
@@ -48,7 +48,7 @@
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.interceptor.RequirePOST;
 
-import javax.annotation.CheckForNull;
```
Comment (basil) [inline src/main/java/com/myyearbook/hudson/plugins/confluence/ConfluenceSite.java:51]: JSR 305 was EOL'ed a long time ago.

**Review by jhansche — APPROVED** [review]
Nice, thanks 👍

**Pull request merged**